### PR TITLE
Tianchang dev support oss

### PR DESCRIPTION
**Pull request by caszkgui**
[rename function timeout in oss](https://github.com/tensorflow/io/issues/238)
[support multi oss cluster in one training job ](https://github.com/tensorflow/io/issues/239)

### CODE REVIEW

File: tests/test_ossfs.py [hunk 1]
```diff
@@ -27,22 +27,23 @@
...
+oss_id = None
+oss_key = None
+oss_host = None
 
-_msg = ("OSS tests skipped. To enable them, set OSS_CREDENTIALS env variable "
-        "to the path of your oss credential file and OSS_FS_TEST_BUCKET env "
-        "variable to your oss test bucket name.")
+_msg = ("OSS tests skipped. To enable them, set oss_id, oss_key, oss_host and bucket variable "
+        "to its real value")
+def check_oss_variable():
```
Comment (terrytangyuan) [inline tests/test_ossfs.py]: Should this be an internal function, e.g. check_oss_variable -> _check_oss_variable?

File: tests/test_ossfs.py [hunk 2]
```diff
@@ -74,36 +75,39 @@ def test_file_operations(self):
...
-    content = gfile.ListDirectory("oss://" + bucket)
-    content_s = gfile.ListDirectory("oss://" + bucket + "/")
+    content = gfile.ListDirectory("oss://%s\x01id=%s\x02key=%s\x02host=%s" %(bucket, oss_id, oss_key, oss_host))
+    content_s = gfile.ListDirectory("oss://%s\x01id=%s\x02key=%s\x02host=%s/" %(bucket, oss_id, oss_key, oss_host))
     self.assertEqual(content, content_s)
     self.assertIn("oss_fs_test", content)
     self.assertIn("oss_fs_test/d1", content)
     self.assertIn("oss_fs_test/d1/d2", content)
 
-    # Test listing test directory with and without trailing '/'
```
Comment (terrytangyuan) [inline tests/test_ossfs.py]: Why are these tests removed?

**Review by terrytangyuan — COMMENTED** [review]
LGTM in general. Just some small comments.

File: tensorflow_io/oss/kernels/ossfs/oss_file_system.cc [hunk 3]
```diff
@@ -570,57 +539,78 @@ Status OSSFileSystem::_ParseOSSURIPath(StringPiece fname, std::string* bucket,
...
+      return errors::InvalidArgument("OSS path does not contain a bucket name:", fname);
     }
 
-    if (access_id_.empty() || access_key_.empty() || host_.empty()) {
-      return errors::InvalidArgument(
-          "OSS path does not contain valid access info:", fname);
-    }
-    if (init_) {
-      init_ = false;
+  }else if (bucketp.find('?') != StringPiece::npos) {
```
Comment (piaoyats) [inline tensorflow_io/oss/kernels/ossfs/oss_file_system.cc]: need space ,code style 
}else  => } else

File: tensorflow_io/oss/kernels/ossfs/oss_file_system.cc [hunk 4]
```diff
@@ -503,63 +496,39 @@ class OSSWritableFile : public WritableFile {
...
+  object = string(remaining);
 
-  if (bucketp.find('?') != StringPiece::npos) {
+  if (bucketp.find('\x01') != StringPiece::npos) {
     // contains id, key, host information
-    size_t pos = bucketp.find('?');
-    *bucket = string(bucketp.substr(0, pos));
-    StringPiece access_info = bucketp.substr(pos + 1);
-    std::vector<std::string> access_infos = str_util::Split(access_info, "&");
+    size_t pos = bucketp.find('\x01');
```
Comment (zou000) [inline tensorflow_io/oss/kernels/ossfs/oss_file_system.cc]: move this to line 254 above and reuse the result in the if condition.

File: tensorflow_io/oss/kernels/ossfs/oss_file_system.cc [hunk 5]
```diff
@@ -503,63 +496,39 @@ class OSSWritableFile : public WritableFile {
...
+    size_t tmpPos = pos;
+    bucket = string(bucketp.substr(0, pos));
+    StringPiece access_info = bucketp.substr(pos+1);
+    std::vector<std::string> access_infos = str_util::Split(access_info, "\x02");
     for (auto& key_value : access_infos) {
```
Comment (zou000) [inline tensorflow_io/oss/kernels/ossfs/oss_file_system.cc]: const auto&

File: tensorflow_io/oss/kernels/ossfs/oss_file_system.cc [hunk 6]
```diff
@@ -570,57 +539,78 @@ Status OSSFileSystem::_ParseOSSURIPath(StringPiece fname, std::string* bucket,
...
 
-    if (access_id_.empty() || access_key_.empty() || host_.empty()) {
-      return errors::InvalidArgument(
-          "OSS path does not contain valid access info:", fname);
-    }
-    if (init_) {
-      init_ = false;
+  }else if (bucketp.find('?') != StringPiece::npos) {
+    // contains id, key, host information
+    size_t pos = bucketp.find('?');
```
Comment (zou000) [inline tensorflow_io/oss/kernels/ossfs/oss_file_system.cc]: similar as above, do search once and reuse the result.

File: tensorflow_io/oss/kernels/ossfs/oss_file_system.cc [hunk 7]
```diff
@@ -1049,47 +1054,185 @@ Status OSSFileSystem::GetFileSize(const std::string& fname, uint64* file_size) {
...
+  std::string host, access_id, access_key;
+  TF_RETURN_IF_ERROR(
+      _ParseOSSURIPath(src, sbucket, sobject, host, access_id, access_key));
   std::string dobject, dbucket;
-  TF_RETURN_IF_ERROR(_ParseOSSURIPath(target, &dbucket, &dobject));
+  TF_RETURN_IF_ERROR(
+      _ParseOSSURIPath(target, dbucket, dobject, host, access_id, access_key));
 
-  OSSConnection oss(host_, access_id_, access_key_);
+  OSSConnection oss(host, access_id, access_key);
```
Comment (zou000) [inline tensorflow_io/oss/kernels/ossfs/oss_file_system.cc:1048]: what happens if source and target have different host/access_id/access_key? shoule we return an error here?

File: tensorflow_io/oss/kernels/ossfs/oss_file_system.cc [hunk 8]
```diff
@@ -1049,47 +1054,185 @@ Status OSSFileSystem::GetFileSize(const std::string& fname, uint64* file_size) {
...
-  aos_str_set(&dest_object, dobject.c_str());
-  headers = aos_table_make(pool, 0);
 
-  s = oss_copy_object(oss_options, &source_bucket, &source_object, &dest_bucket,
-                      &dest_object, headers, &resp_headers);
+  Status status = IsDirectory(src);
+  if (status.ok()) {
+    std::vector<std::string> childPaths;
+    _ListObjects(pool, oss_options, sbucket, sobject, &childPaths, true, false, false, 1000);
+    for (auto child : childPaths) {
```
Comment (zou000) [inline tensorflow_io/oss/kernels/ossfs/oss_file_system.cc]: const auto&

File: tensorflow_io/oss/kernels/ossfs/oss_file_system.cc [hunk 9]
```diff
@@ -12,17 +12,16 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
 #include <unistd.h>
 #include <cstdlib>
 #include <ctime>
 #include <fstream>
 #include <iostream>
 #include <vector>
+#include <cmath>
 
 #include "aos_string.h"
-#include "minIni.h"
```
Comment (zou000) [inline tensorflow_io/oss/kernels/ossfs/oss_file_system.cc:25]: Now we don't require minIni, could delete third_party/minini.BUILD. Also modify WORKSPACE to remove the corresponding http_archive

**Review by zou000 — CHANGES_REQUESTED** [review]

File: tensorflow_io/oss/kernels/ossfs/oss_file_system.cc (hunk 7 above)
Comment (caszkgui) [reply tensorflow_io/oss/kernels/ossfs/oss_file_system.cc:1048]: yes, we should check whether source and target have the same host/access_id/access_key first, and return an error if not.

File: tensorflow_io/oss/kernels/ossfs/oss_file_system.cc (hunk 9 above)
Comment (caszkgui) [reply tensorflow_io/oss/kernels/ossfs/oss_file_system.cc:25]: additionally, we remove minini dependency from tensorflow/oss/BUILD

**Pull request closed without merging**